src/util/__tests__/createServerRootMixin.test.js

```diff
@@ -336,187 +374,191 @@ Array [
             },
           },
         },
-        render(h) {
-          return h(InstantSearchSsr, {}, [
-            h(Configure, {
-              attrs: {
-                hitsPerPage: 100,
-              },
-            }),
+        render: renderCompat(h =>
+          h(InstantSearchSsr, {}, [
+            h(
+              Configure,
+              isVue3
+                ? { hitsPerPage: 100 }
+                : {
+                    attrs: {
+                      hitsPerPage: 100,
+                    },
+                  }
+            ),
             h(SearchBox),
-          ]);
-        },
+          ])
+        ),
         serverPrefetch() {
           return this.instantsearch.findResultsState(this);
         },
-      });
+      };
 
-      const wrapper = new Vue({
+      const wrapper = createSSRApp({
         mixins: [forceIsServerMixin],
-        render(h) {
-          return h(App, { props: { someProp } });
-        },
+        render: renderCompat(h =>
+          h(App, isVue3 ? { someProp } : { props: { someProp } })
+        ),
```

Comment:
Little change in the syntax. Because of things like this, in general, it's easier (and more readable) to write in template form rather than in render function.

Reply:
I like renderCompat, could we use it everywhere (InstantSearch/configure component too)? maybe with these conditionals already part of the item?

Reply:
Good idea. I will do it in a separate PR. Just to clarify,

> with these conditionals already part of the item?

What do you mean by "item"? You mean the createElement, h, and these conditional props altogether covered by renderCompat function?